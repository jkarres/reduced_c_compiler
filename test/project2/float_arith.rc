
function : void main() {
  float e = 2.71828183;
  const float pi = 3.14159265;
  int i = 2;
  const int j = 3;

  cout << 1 << " " << 11.0 << endl;
  cout << 2 << " " << +11.0 << endl;
  cout << 3 << " " << -11.0 << endl;

  cout << 4 << " " << pi << endl;
  cout << 5 << " " << +pi << endl;
  cout << 6 << " " << -pi << endl;

  cout << 7 << " " << e << endl;
  cout << 8 << " " << +e << endl;
  cout << 9 << " " << -e << endl;

  cout << 10 << " " << e + pi << endl;
  cout << 11 << " " << e - pi << endl;
  cout << 12 << " " << e * pi << endl;
  cout << 13 << " " << e / pi << endl;
  cout << 14 << " " << pi + e << endl;
  cout << 15 << " " << pi - e << endl;
  cout << 16 << " " << pi * e << endl;
  cout << 17 << " " << pi / e << endl;

  cout << 18 << " " << i + pi << endl;
  cout << 19 << " " << pi + i << endl;
  cout << 20 << " " << i - pi << endl;
  cout << 21 << " " << pi - i << endl;
  cout << 22 << " " << i * pi << endl;
  cout << 23 << " " << pi * i << endl;
  cout << 24 << " " << i / pi << endl;
  cout << 25 << " " << pi / i << endl;

  cout << 26 << " " << j + pi << endl;
  cout << 27 << " " << pi + j << endl;
  cout << 28 << " " << j - pi << endl;
  cout << 29 << " " << pi - j << endl;
  cout << 30 << " " << j * pi << endl;
  cout << 31 << " " << pi * j << endl;
  cout << 32 << " " << j / pi << endl;
  cout << 33 << " " << pi / j << endl;

  cout << 34 << " " << i + e << endl;
  cout << 35 << " " << e + i << endl;
  cout << 36 << " " << i - e << endl;
  cout << 37 << " " << e - i << endl;
  cout << 38 << " " << i * e << endl;
  cout << 39 << " " << e * i << endl;
  cout << 40 << " " << i / e << endl;
  cout << 41 << " " << e / i << endl;

  cout << 42 << " " << j + e << endl;
  cout << 43 << " " << e + j << endl;
  cout << 44 << " " << j - e << endl;
  cout << 45 << " " << e - j << endl;
  cout << 46 << " " << j * e << endl;
  cout << 47 << " " << e * j << endl;
  cout << 48 << " " << j / e << endl;
  cout << 49 << " " << e / j << endl;

}
