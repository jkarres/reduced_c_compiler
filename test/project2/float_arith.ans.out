1 11.00
2 11.00
3 -11.00
4 3.14
5 3.14
6 -3.14
7 2.72
8 2.72
9 -2.72
10 5.86
11 -0.42
12 8.54
13 0.87
14 5.86
15 0.42
16 8.54
17 1.16
18 5.14
19 5.14
20 -1.14
21 1.14
22 6.28
23 6.28
24 0.64
25 1.57
26 6.14
27 6.14
28 -0.14
29 0.14
30 9.42
31 9.42
32 0.95
33 1.05
34 4.72
35 4.72
36 -0.72
37 0.72
38 5.44
39 5.44
40 0.74
41 1.36
42 5.72
43 5.72
44 0.28
45 -0.28
46 8.15
47 8.15
48 1.10
49 0.91
